docs/tutorial/part_2.rst

```diff
@@ -0,0 +1,459 @@
+instances to talk with each other, and with other parts of Django.
+
+A channel layer provides the following abstractions:
+
+* A **channel** is a mailbox where messages can be sent to. Each channel has a name.
+  Anyone who has the name of a channel can send a message to the channel.
+
+* A **group** is a group of related channels. A group has a name. Anyone who has the
+  name of a group can (1) add/remove a channel to the group by name and (2) send
+  a message to all channels in the group. It is not possible to enumerate what
+  channels are in a particular group.
+
+Every consumer instance has an automatically generated unique channel name, and
+so can be communicated with via a channel layer.
+
+In our chat application we want to have multiple instances of ``ChatConsumer`` in
+the same room communicate with each other. To do that we will have each
+ChatConsumer add its channel to a group whose name is based on the room name.
+That will allow ChatConsumers to transmit messages to all other ChatConsumers in
+the same room.
+
+We will use a channel layer that uses Redis as its backing store. To start a
+Redis server on port 6379, run the following command::
+
+    $ docker run -p 6379:6379 -d redis:2.8
+
+We need to install channels_redis so that Channels knows how to interface with
+Redis. Run the following command::
+
+    $ pip3 install channels_redis
+
+Before we can use a channel layer, we must configure it. Edit the
+**mysite/settings.py** file and add a ``CHANNEL_LAYERS`` setting to the bottom.
+It should look like::
+
+    # mysite/settings.py
+    # Channels
+    ASGI_APPLICATION = 'mysite.routing.application'
+    CHANNEL_LAYERS = {
+        'default': {
+            'BACKEND': 'channels_redis.core.RedisChannelLayer',
+            'CONFIG': {
+                "hosts": [('127.0.0.1', 6379)],
+            },
+        },
+    }
+
+.. note::
+    It is possible to have multiple channel layers configured.
+    However most projects will just use a single ``'default'`` channel layer.
+
+Let’s make sure that the channel layer can communicate with Redis. Open a Django
+shell and run the following commands::
+
+    $ python3 manage.py shell
+    >>> import channels.layers
+    >>> channel_layer = channels.layers.get_channel_layer()
+    >>> from asgiref.sync import async_to_sync
+    >>> async_to_sync(channel_layer.send)('test_channel', {type: 'hello'})
+    >>> async_to_sync(channel_layer.receive)('test_channel')
+    {'type': 'hello'}
+
+Type Control-D to exit the Django shell.
+
+Now that we have a channel layer, let’s use it in ``ChatConsumer``. Put the
+following code in **chat/consumers.py**, replacing the old code::
+
+    # chat/consumers.py
+    from asgiref.sync import async_to_sync
+    from channels.generic.websocket import WebsocketConsumer
+    import json
+    
+    class ChatConsumer(WebsocketConsumer):
+        def connect(self):
+            self._room_name = self.scope['url_route']['kwargs']['room_name']
```

Comment:
I don't think there's any need to use `_room_name` rather than just `room_name` here. I'd like to reinforce that it's just a normal attribute on `self`.